### ISSUE TITLE
Don't support the latest flymake

### ISSUE THREAD

**Issue by xuchunyang** [tam17aki/helm-flymake#4]
I encounter an error when I type `M-x helm-flymake`, here is the backtrace

> Debugger entered--Lisp error: (void-variable flymake-err-info)
>   helm-flymake-get-err-list()
>   helm-flymake-get-err-list-sorted()
>   helm-flymake-get-candidate("w")
>   helm-flymake-init("w")
>   (lambda nil (helm-flymake-init "w"))()
>   apply((lambda nil (helm-flymake-init "w")) nil)
>   helm-funcall-with-source(((name . "Flymake Warning") (init lambda nil (helm-flymake-init "w")) (candidates-in-buffer) (candidate-transformer lambda (cands) (delete "" cands)) (type . line) (action ("Goto Line" . helm-flymake-action-goto-line)) (recenter)) (lambda nil (helm-flymake-init "w")))
>   helm-funcall-foreach(init)
>   helm-initial-setup(nil)
>   helm-initialize(nil nil nil (helm-source-flymake-warning helm-source-flymake-error))
>   helm-internal((helm-source-flymake-warning helm-source-flymake-error) nil nil nil nil #<buffer *helm flymake*> nil nil nil)
>   apply(helm-internal ((helm-source-flymake-warning helm-source-flymake-error) nil nil nil nil #<buffer *helm flymake*> nil nil nil))
>   helm((helm-source-flymake-warning helm-source-flymake-error) nil nil nil nil #<buffer *helm flymake*> nil nil nil)
>   apply(helm ((helm-source-flymake-warning helm-source-flymake-error) nil nil nil nil #<buffer *helm flymake*> nil nil nil))
>   helm(:sources (helm-source-flymake-warning helm-source-flymake-error) :buffer #<buffer *helm flymake*> :input nil)
>   helm-flymake(nil)
>   funcall-interactively(helm-flymake nil)
>   call-interactively(helm-flymake record nil)
>   command-execute(helm-flymake record)
>   helm-M-x(nil #("helm-flymake" 0 12 (match-part "helm-flymake")))
>   funcall-interactively(helm-M-x nil #("helm-flymake" 0 12 (match-part "helm-flymake")))
>   call-interactively(helm-M-x nil nil)
>   command-execute(helm-M-x)
> 

I built my Emacs from git repo recently. The variable `flymake-err-info` was removed  a few months ago in https://github.com/emacs-mirror/emacs/commit/94a88c1ae98b1bf7ab80ae9cdd4d6e16b36597ef. I think I can fix this issue when I have time and get familiar with flymake.

**Comment by manuel-uberti** [comment]
Any update on this? :)

**Comment by xuchunyang** [comment]
Nope. 

But I just tried helm-flymake again using Emacs 27.1, and it fails because of another issue:

> helm-M-x-execute-command: Symbol's value as variable is void: flymake-err-info

I also got some byte compiler warnings during M-x package-install

> In toplevel form:
> helm-flymake.el:85:30:Warning: Package cl is deprecated
> 
> In helm-flymake-get-err-list:
> helm-flymake.el:92:25:Warning: reference to free variable ‘flymake-err-info’
> 
> In helm-flymake-get-candidate:
> helm-flymake.el:106:6:Warning: ‘mapcar’ called for effect; use ‘mapc’ or
>     ‘dolist’ instead
> 
> In end of data:
> helm-flymake.el:187:1:Warning: the following functions are not known to be
>     defined: flymake-ler-line, flymake-ler-type, flymake-ler-text

I am not sure if helm-flymake works with latest flymake.

**Comment by manuel-uberti** [comment]
Yes, there is no `flymake-err-info` any more, the API has changed quite a lot.

**Comment by zbelial** [comment]
Hi guys, it works with the latest flymake now. Enjoy it.
